File: typescript/src/main/vsdk/toolkit/raycasting/stochasticRaytracing/McradP.ts
import { ColorRgb } from "../../common/color/ColorRgb";
import { Patch } from "../../environment/geometry/elements/Patch";
import { GalerkinBasis } from "./GalerkinBasis";
type StochasticRadiosityElement = any;

export class McradP {
  private constructor() {
  }

  public static numberOfVertices(elem: StochasticRadiosityElement): number {
    return elem.patch!.numberOfVertices;
  }

  public static topLevelStochasticRadiosityElement(patch: Patch): StochasticRadiosityElement {
    return patch.radianceData as unknown as StochasticRadiosityElement;
  }

  public static getTopLevelPatchRad(patch: Patch): ColorRgb[] | null {
    return McradP.topLevelStochasticRadiosityElement(patch).radiance;
  }

  public static getTopLevelPatchUnShotRad(patch: Patch): ColorRgb[] | null {
    return McradP.topLevelStochasticRadiosityElement(patch).unShotRadiance;
  }

  public static getTopLevelPatchReceivedRad(patch: Patch): ColorRgb[] | null {
    return McradP.topLevelStochasticRadiosityElement(patch).receivedRadiance;
  }

  public static getTopLevelPatchBasis(patch: Patch): GalerkinBasis | null {
    return McradP.topLevelStochasticRadiosityElement(patch).basis;
  }
}

File: java/src/main/java/vsdk/toolkit/raycasting/stochasticRaytracing/McradP.java
package vsdk.toolkit.raycasting.stochasticRaytracing;

import vsdk.toolkit.common.color.ColorRgb;
import vsdk.toolkit.environment.geometry.elements.Patch;

public final class McradP {
    private McradP() {
    }

    public static int numberOfVertices(StochasticRadiosityElement elem) {
        return elem.patch.numberOfVertices;
    }

    public static StochasticRadiosityElement topLevelStochasticRadiosityElement(Patch patch) {
        return (StochasticRadiosityElement)patch.radianceData;
    }

    public static ColorRgb[] getTopLevelPatchRad(Patch patch) {
        return topLevelStochasticRadiosityElement(patch).radiance;
    }

    public static ColorRgb[] getTopLevelPatchUnShotRad(Patch patch) {
        return topLevelStochasticRadiosityElement(patch).unShotRadiance;
    }

    public static ColorRgb[] getTopLevelPatchReceivedRad(Patch patch) {
        return topLevelStochasticRadiosityElement(patch).receivedRadiance;
    }

    public static GalerkinBasis getTopLevelPatchBasis(Patch patch) {
        return topLevelStochasticRadiosityElement(patch).basis;
    }
}
